authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -665,8 +690,43 @@ public Boolean onMetadataUpdated(MetadataObjectChange... changes) throws Runtime
           // Did not need to update the Ranger policy
           continue;
         }
-        List<AuthorizationMetadataObject> oldAuthzMetadataObjects =
-            translateMetadataObject(metadataObject);
+
+        // If locations don't change, we don't need to modify the policies
+        if (renameChange.locations() == null || renameChange.locations().isEmpty()) {
+          continue;
+        }
```

Comment:
I think maybe we need to check if locations have changed?

Reply:
I should add comments to explain this. This means fileset , topic won't modify the locations. We don't need change the locations.

Reply:
Changed.